ivy_tests/test_ivy/helpers.py

```diff
@@ -519,6 +524,82 @@ def value_test(*, ret_np_flat, ret_from_np_flat, rtol=None, atol=1e-6):
             assert_all_close(ret_np, ret_from_np, rtol=rtol, atol=atol)
 
 
+def args_to_container(array_args):
+    array_args_container = ivy.Container({str(k): v for k, v in enumerate(array_args)})
+    return array_args_container
+
+
+def gradient_test(
+    *,
+    fn_name,
+    all_as_kwargs_np,
+    args_np,
+    kwargs_np,
+    input_dtypes,
+    as_variable_flags,
+    native_array_flags,
+    container_flags,
+    rtol_: float = None,
+    atol_: float = 1e-06,
+    ground_truth_backend: str = "torch",
+):
+    def grad_fn(xs):
+        array_vals = [v for k, v in xs.to_iterator()]
+        args_writeable = ivy.copy_nest(args)
+        kwargs_writeable = ivy.copy_nest(kwargs)
+        ivy.set_nest_at_indices(args_writeable, args_idxs, array_vals)
```

Comment:
we are setting the exact same array values into both the `args` and the `kwargs`, this is not correct.

Reply:
Yeah, I made a mistake here too, I've fixed it now.